tilecache > DEFAULT_TILE_CACHE_HIGH_WATER_MARK will show error

ol.js show Uncaught AssertionError: Assertion failed
when tilecahcer > ol.DEFAULT_TILE_CACHE_HIGH_WATER_MARK

![default](https://cloud.githubusercontent.com/assets/8294943/6524441/74c6b080-c434-11e4-94e6-e4bcc8e77286.PNG)

i change ol.DEFAULT_TILE_CACHE_HIGH_WATER_MARK=20
http://fiddle.jshell.net/s2evvztt/
open fiddle and zoom in 
or use ol.js and load tilecache over 2048 (ol.DEFAULT_TILE_CACHE_HIGH_WATER_MARK = 2048;)

--- a/src/ol/imagetile.js
+++ b/src/ol/imagetile.js
@@ -68,7 +68,9 @@ goog.inherits(ol.ImageTile, ol.Tile);
  * @inheritDoc
  */
 ol.ImageTile.prototype.disposeInternal = function() {
-  this.unlistenImage_();
+  if (this.state == ol.TileState.LOADING) {
+    this.unlistenImage_();
+  }
   goog.base(this, 'disposeInternal');
 };
 

